src/LensPlatformClient.ts

```diff
@@ -45,6 +45,25 @@ interface DecodedAccessToken {
   typ: string;
 }
 
+/**
+ * TypeGuard to determine if type is GotRequestFunction
+ */
+const isGotRequestFunction = (
+  func: any, key: keyof Got
+): func is GotRequestFunction =>
+  typeof func === "function" &&
+  ["get", "post", "put", "patch", "head", "delete"].includes(key);
```

Comment:
This could be written to be slightly more type safe:
```
const gotMethods: Array<keyof Got> = ["get", "post", "put", "patch", "head", "delete"];

/**
 * TypeGuard to determine if type is GotRequestFunction
 */
const isGotRequestFunction = (
  func: any, key: keyof Got
): func is GotRequestFunction =>
  typeof func === "function" && gotMethods.includes(key);
```

Sadly that would allow developer to forgot to include some method, but at least unknown methods would error.